tools/tests/py2fgen/test_export.py

```diff
@@ -6,5 +6,53 @@
 # Please, refer to the LICENSE file in the root directory.
 # SPDX-License-Identifier: BSD-3-Clause
 
+from typing import Annotated, Any
 
-# TODO
+import numpy as np
+import pytest
+
+from icon4py.tools import py2fgen
+from icon4py.tools.py2fgen import _export
+
+
+def test_from_annotated():
+    testee = Annotated[int, py2fgen.ScalarParamDescriptor(dtype=py2fgen.INT32)]
+
+    result = _export._from_annotated(testee)
+
+    assert isinstance(result, py2fgen.ScalarParamDescriptor)
+    assert result.dtype == py2fgen.INT32
+
+
+@pytest.mark.parametrize(
+    "testee, expected",
+    [
+        (float, py2fgen.ScalarParamDescriptor(dtype=py2fgen.FLOAT32)),
+        (np.float32, py2fgen.ScalarParamDescriptor(dtype=py2fgen.FLOAT32)),
+        (int, pytest.raises),  # no descriptor deducible
```

Comment:
You could also pass here the expected exception type, instead of hardcoding it in the test function
```suggestion
        (int, ValueError),  # no descriptor deducible
```

Reply:
see my changes, do you have a more elegant way?